Fixes #22836, #23325: disable actions when necessary.

waldenraines opened the pull request:
If we are in disconnected mode or a task is in progress we need to
disable ability to add subscriptions, delete subscriptions,
edit/modify subscriptions, and refresh manifest.

Also fixes the manifest modal table after the changes to indices.

http://projects.theforeman.org/issues/22836
http://projects.theforeman.org/issues/23325

TODO:
- [x] Add missing tests

waldenraines commented:
> waldenraines deleted a comment from houndci-bot just now

Hound was stuck on something being a problem even though it had been fixed.

tstrachota commented on webpack/move_to_foreman/components/DisabledButtonWithTooltip.js:
```diff
@@ -0,0 +1,51 @@
+
+  return (
+    <span className="btn-container">
+      {disabled ? getTooltipButton() : getButton()}
```
Reading the code and readme of the tooltip button package, aren't you duplicating its functionality here?
According to https://github.com/pastaclub/react-bootstrap-tooltip-button I think you can use the component directly.

waldenraines replied:
Oops, that was left over from a previous implementation.  I realized when looking into this that I could just delete this entire component and use the tooltip button directly so I did.

tstrachota commented on webpack/scenes/Subscriptions/Manifest/ManifestHistoryTableSchema.js:
```diff
@@ -6,6 +6,9 @@ export const columns = [
       label: __('Status'),
       formatters: [headerFormat],
+      props: {
+        index: 0,
```
What are the indexes good for?

waldenraines replied:
They're needed after the changes to the table in https://github.com/Katello/katello/pull/7331.  Basically to support selecting rows.

tstrachota commented on webpack/scenes/Subscriptions/SubscriptionsPage.js:
```diff
@@ -118,30 +129,44 @@ class SubscriptionsPage extends Component {
+                          id="add-subscriptions-button"
+                          disabled={disableManifestActions}
+                          disabledReason={disabledReason}
+                          bsStyle="primary"
```
Is the `bsStyle` and `id` actually propagated to the button? The props doesn't seem to be passed down.

tstrachota replied:
Update: I see where the `id` goes now. But `bsStyle` is ignored.

waldenraines replied:
Good catch, will fix.

waldenraines commented:
@tstrachota fixed comments, rebased, and added tests.

johnpmitsch commented:
getting this when clicking manage manifest (no idea if this was introduced by you or not)
```
warning.js:33 Warning: Each child in an array or iterator should have a unique "key" prop.

Check the render method of `Table`. See https://fb.me/react-warning-keys for more information.
    in Header (created by Table)
    in Table (created by ManageManifestModal)
    in Spinner (created by ManageManifestModal)
    in div (created by TabPane)
    in Transition (created by Fade)
    in Fade (created by TabPane)
    in TabPane (created by Tab)
    in Tab (created by ManageManifestModal)
    in div (created by TabContent)
    in TabContent (created by Tabs)
    in div (created by Tabs)
    in TabContainer (created by Tabs)
    in Tabs (created by Uncontrolled(Tabs))
    in Uncontrolled(Tabs) (created by ManageManifestModal)
    in div (created by ModalBody)
    in ModalBody (created by ManageManifestModal)
    in div (created by CustomModalDialog)
    in div (created by CustomModalDialog)
    in div (created by CustomModalDialog)
    in CustomModalDialog (created by Modal)
    in Transition (created by Fade)
    in Fade (created by Modal)
    in RefHolder (created by Modal)
    in div (created by Modal)
    in Portal (created by Modal)
    in Modal (created by Modal)
    in Modal (created by ManageManifestModal)
    in ManageManifestModal (created by Connect(ManageManifestModal))
    in Connect(ManageManifestModal) (created by SubscriptionsPage)
    in div (created by Col)
    in Col (created by SubscriptionsPage)
    in div (created by Row)
    in Row (created by SubscriptionsPage)
    in div (created by Grid)
    in Grid (created by SubscriptionsPage)
    in SubscriptionsPage (created by Connect(SubscriptionsPage))
    in Connect(SubscriptionsPage) (created by Route)
    in Route
    in div
    in Unknown (created by Application)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by Application)
    in Application (created by Connect(Application))
    in Connect(Application)
    in Provider
```

johnpmitsch commented:
Everything works as expected :+1: just one question around the error

waldenraines commented:
> Everything works as expected  just one question around the error

Do you know if you get that error in master?

johnpmitsch commented:
@waldenraines I do see this error in master, and looks like the page may be broken by my changes, please rebase and test this out

waldenraines commented:
> @waldenraines I do see this error in master, and looks like the page may be broken by my changes, please rebase and test this out

Rebased and fixed this issue.

waldenraines commented:
[katello test]

waldenraines commented:
[test katello]

waldenraines commented:
[test katello]

waldenraines commented:
[test katello]

waldenraines commented:
> Works well. Since the previous mentioned error is in master, we can address that seperately

See http://projects.theforeman.org/issues/23552

johnpmitsch commented:
@tstrachota shouldn't we fix these issues in pf-react? It seems like the formatters should be able to be used as intended

waldenraines commented:
[test katello]

tstrachota commented:
> Thoughts on opening a PR after this is merged?

Np, I just wanted to let you know that I've fixed that.
